### PR TITLE
Allow multiple vectorfunctionspace subcomponents in pointwise assembly

### PR DESCRIPTION
**Pull request by wence-**
This allows `z.assign(x - y)` where the space is mixed and you have multiple VFS bits.

Doesn't support (but didn't previously) the case where you have VFS(RT) [say].

### CODE REVIEW

**Review by pefarrell — APPROVED** [review]
Looks good. Fixes our bug with mixed assignment. Thanks!

File: firedrake/assemble_expressions.py [hunk 1]
```diff
@@ -158,19 +180,25 @@ def _split_component_tensor(o, self):
...
-        else:
+        elif not inct:
+            idx = indices[:ncmp]
+            indices = indices[ncmp:]
             mi = multiindex if multiindex.indices() == idx else MultiIndex(idx)
             result.append(ufl_reuse_if_untouched(o, agg, mi))
+        else:
+            # shape and inct
+            aggshape = (flatten(agg.ufl_shape)
+                        + tuple(itertools.repeat(1, len(aggregate.ufl_shape) - 1)))
```
Comment (ksagiyam) [inline firedrake/assemble_expressions.py:199]: Do we ever get here with `len(aggregate.ufl_shape) != 1`?

Comment (wence-) [reply firedrake/assemble_expressions.py:199]: I suspect I did it for belt-and-braces.

Comment (wence-) [reply firedrake/assemble_expressions.py:199]: I guess know because `split_coefficient` flattens.

**Pull request merged**